rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/monitoring/views/compact/CompactView.java

```diff
@@ -0,0 +1,312 @@
+                changeNodeSourceStatusIfChanged(flow, nodeSource);
+            }
+        }
+    }
+
+    private void processNodes(CompactFlowPanel flow, List<Node> nodes) {
+        for (Node node : nodes) {
+            if (node.isRemoved()) {
+                removeNode(flow, node);
+            } else {
+                addNodeIfNotExists(flow, node);
+                changeNodeStatusIfChanged(flow, node);
+            }
+        }
+    }
+
+    private void removeNode(CompactFlowPanel flow, Node node) {
+        if (flow.isNodeSourceDrawn(node.getSourceName())) {
+            flow.remove(node);
+        }
+    }
+
+    private void changeNodeStatusIfChanged(CompactFlowPanel flow, Node node) {
+        if (node.isChanged()) {
+            flow.redrawNode(node);
+        }
+    }
+
+    private void addNodeIfNotExists(CompactFlowPanel flow, Node node) {
+        if (!flow.isNodeDrawn(node)) {
+            Tile nodeTile = new Tile(this, flow, node);
+            Tile hostTile = new Tile(this, flow, new Host(node.getHostName(), node.getSourceName()));
+            if (node.isVirtual()) {
+                hostTile.getHost().setVirtual(true);
+            }
+            flow.drawNode(nodeTile, hostTile);
+        }
+    }
+
+    private void changeNodeSourceStatusIfChanged(CompactFlowPanel flow, NodeSource nodeSource) {
+        if (nodeSource.isChanged()) {
+            flow.redrawNodeSource(nodeSource);
+        }
+    }
+
+    private void addNodeSourceIfNotExists(CompactFlowPanel flow, NodeSource nodeSource) {
+        if (!flow.isNodeSourceDrawn(nodeSource.getSourceName())) {
+            Tile nsTile = new Tile(this, flow, nodeSource);
+            flow.drawNodeSource(nsTile);
+        }
+    }
+
+    private void removeNodeSource(CompactFlowPanel flow, NodeSource nodeSource) {
+        if (flow.isNodeSourceDrawn(nodeSource.getSourceName())) {
+            flow.remove(nodeSource);
+        }
+    }
+
+    private void initializePanel() {
+        compactPanel = new CompactFlowPanel();
+        myNodesCompactPanel = new CompactFlowPanelOwn();
+
+        campactPanelCanvas = new WidgetCanvas(compactPanel);
+        myNodesCompactPanelCanvas = new WidgetCanvas(myNodesCompactPanel);
+
+        root.addMember(campactPanelCanvas);
+
+        addResizeHandler(compactPanel);
+
+        addResizeHandler(myNodesCompactPanel);
+    }
+
+    private void addResizeHandler(CompactFlowPanel panel) {
+        root.addResizedHandler(event -> {
+            panel.setPixelSize(root.getWidth() - root.getScrollbarSize(), root.getHeight() - root.getScrollbarSize());
+            // lazy hack to force this.compactPanel to -really- relayout
+            root.setBorder(_borderSwitch ? "1px solid white" : "0px");
+            _borderSwitch = !_borderSwitch;
+        });
+    }
+
+    public boolean isDoNotScroll() {
+        return _doNotScroll;
+    }
+
+    public void setDoNotScroll(boolean _doNotScroll) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Rename this local variable to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00117)